### ISSUE TITLE
how does output to /dev/null work

### ISSUE THREAD

**Issue by zummuz** [yt-dlp/yt-dlp#3949]
### Checklist

- [X] I'm asking a question and **not** reporting a bug/feature request
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions including closed ones

### Question

Hi.
Is `-o '/dev/null'`a  valid argument on linux? If I recollect right, it was not letting a video to be saved on a file system, but a few linux distro updates later I have tried it again and it either gives the [Errno 22] or do save a video renaming it to `null.ytdl and null.webm` for other videos. 
Is this new supposed behavior or is it due to an external change?

### Verbose log

```shell
[debug] Command-line config: ['https://www.youtube.com/watch?v=ZavwtJxmqJI', '-f', '299', '-o', '/dev/null', '-Uv']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355]
[debug] Python version 3.10.4 (CPython 64bit) - Linux-5.17.9-1-MANJARO-x86_64-with-glibc2.35
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs
[debug] exe versions: ffmpeg 5.0 (setts), ffprobe 5.0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.12.0, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[debug] [youtube] Extracting URL: https://www.youtube.com/watch?v=ZavwtJxmqJI
[youtube] ZavwtJxmqJI: Downloading webpage
[youtube] ZavwtJxmqJI: Downloading android player API JSON
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
ERROR: [youtube] ZavwtJxmqJI: Requested format is not available. Use --list-formats for a list of available formats
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1395, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1479, in __extract_info
    return self.process_ie_result(ie_result, download, extra_info)
  File "/usr/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1535, in process_ie_result
    ie_result = self.process_video_result(ie_result, download=download)
  File "/usr/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 2622, in process_video_result
    raise ExtractorError(
yt_dlp.utils.ExtractorError: [youtube] ZavwtJxmqJI: Requested format is not available. Use --list-formats for a list of available formats
```

**Comment by zummuz** [comment]
> The log doesn't match the description!

yeah, my bad. sorry

[debug] Command-line config: ['https://www.youtube.com/watch?v=hA55WshsKvg', '-f', '299', '-o', '/dev/null', '-Uv']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355]
[debug] Python version 3.10.4 (CPython 64bit) - Linux-5.17.9-1-MANJARO-x86_64-with-glibc2.35
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs
[debug] exe versions: ffmpeg 5.0 (setts), ffprobe 5.0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.12.0, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[debug] [youtube] Extracting URL: https://www.youtube.com/watch?v=hA55WshsKvg
[youtube] hA55WshsKvg: Downloading webpage
[youtube] hA55WshsKvg: Downloading android player API JSON
[youtube] hA55WshsKvg: Downloading player 7eff7529
[debug] Saving youtube-nsig.7eff7529 to cache
[debug] [youtube] Decrypted nsig yPRtpC8ZiSFYSZPf => tmGPPiOjLH0kGw
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[info] hA55WshsKvg: Downloading 1 format(s): 299
[debug] Invoking http downloader on "https://rr1---sn-xguxaxjvh-8v1s.googlevideo.com/videoplayback?expire=1654255067&ei=e5mZYrTmHcbi7QSiirKACQ&ip=5.18.181.175&id=o-ALObf6ffraIqhmm6zt1OsKSWrGJhJHrSpJ0MjYSzmyJD&itag=299&source=youtube&requiressl=yes&mh=GR&mm=31%2C29&mn=sn-xguxaxjvh-8v1s%2Csn-axq7sn7z&ms=au%2Crdu&mv=m&mvi=1&pcm2cms=yes&pl=24&initcwndbps=1216250&vprv=1&mime=video%2Fmp4&gir=yes&clen=56690849196&dur=111516.197&lmt=1608888423378797&mt=1654233182&fvip=18&keepalive=yes&fexp=24001373%2C24007246&c=ANDROID&txp=6416222&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cgir%2Cclen%2Cdur%2Clmt&sig=AOq0QJ8wRQIgNVMRjDRW3y45-IUDOQEYDNKw-bNeDOlkUwZqQw8sx7ICIQD3l8n0m7GKiu96oyQrF3RnsJI-9wDcIb35vLFATT3AjQ%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpcm2cms%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRgIhANhGV17PX6DpRlV5XOOI2dtVJsOAY9TW-x_CnmX1eYQmAiEAwBRXAyCbGPrDBjtP80qAqIBjwZfr2YyU5YNJWwmQUt0%3D"
[download] Unable to open file due to file access error. Retrying (attempt 1 of 3) ...
[download] Unable to open file due to file access error. Retrying (attempt 2 of 3) ...
[download] Unable to open file due to file access error. Retrying (attempt 3 of 3) ...
ERROR: unable to open for writing: [Errno 22] Invalid argument
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/yt_dlp/downloader/http.py", line 267, in download
    ctx.stream, ctx.tmpfilename = self.sanitize_open(
  File "/usr/lib/python3.10/site-packages/yt_dlp/downloader/common.py", line 224, in inner
    return func(self, *args, **kwargs)
  File "/usr/lib/python3.10/site-packages/yt_dlp/downloader/common.py", line 241, in sanitize_open
    f, filename = sanitize_open(filename, open_mode)
  File "/usr/lib/python3.10/site-packages/yt_dlp/utils.py", line 621, in sanitize_open
    stream = locked_file(filename, open_mode, block=False).__enter__()
  File "/usr/lib/python3.10/site-packages/yt_dlp/utils.py", line 2055, in __enter__
    raise e
  File "/usr/lib/python3.10/site-packages/yt_dlp/utils.py", line 2052, in __enter__
    self.f.truncate()
OSError: [Errno 22] Invalid argument

**Comment by gamer191** [comment]
> It doesn't seem like /dev/nul ever worked, even with youtube-dl

You left out an l in /dev/null

**Comment by pukkandan** [comment]
ah, my bad

**Comment by pukkandan** [comment]
Fixed in 8a82af3511b4379af0d239dbd01c672c17a2c46a
